### PR TITLE
[Admin Portal Revamp] Microgateway Labels

### PR DESCRIPTION
**Pull request by chathuranga95**
### Listing of MG-Labels
listing

### Add/Edit MG Label
editing

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { createMuiTheme } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+
+let id;
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:9]: Why we need to define a variable here ?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+theme.spacing(2);
+
+const InputList = (props) => {
+    const {
+        onInputListChange, initialList, inputLabelPrefix, helperText, addButtonLabel,
+    } = props;
+    const [userInputItems, setUserInputItems] = useState([]);
+
+    const handleInput = (e) => {
+        let tempItems = userInputItems.filter((item) => item.key !== e.target.name);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:21]: Please destructure the event values before use

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        for (let i = 0; i < userInputItems.length; i++) {
+            if (userInputItems[i].value && userInputItems[i].value.trim() !== '') {
+                nonEmptyItems.push(userInputItems[i].value.trim());
+            }
+        }
+        onInputListChange(nonEmptyItems);
+    }, [userInputItems]);
+
+    useEffect(() => {
+        id = 1;
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:45]: Have we ignored eslint check in this file ? , AFAIR this is an eslint rule violation

Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:45]: eslint check is enabled for this file.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                            value={item.value}
+                            helperText={helperText}
+                            variant='outlined'
+                        />
+                        <Box mt={1}>
+                            <Button
+                                color='primary'
+                                onClick={() => handleDelete(item.key)}
+                            >
+                                    Remove
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx:80]: Add i18n

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+            name: originalName,
+            description: originalDescription,
+            hosts: originalHosts,
+        };
+    }
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const { name, description, hosts } = state;
+
+
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx:95]: Remove unnecessary empty lines

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/DeleteMGLabel.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    };
+
+    return (
+        <FormDialogBase
+            title='Delete Microgateway Label?'
+            saveButtonText='Delete'
+            icon={<DeleteForeverIcon />}
+            formSaveCallback={formSaveCallback}
+        >
+            <DialogContentText>Are you sure you want to delete this Microgateway Label?</DialogContentText>
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/DeleteMGLabel.jsx:66]: Add i18n

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx (hunk 1 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8516/commits/c285a00fc8cc9c25a2bd9a0cc4fcc97187a91e15

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx (hunk 2 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8516/commits/c285a00fc8cc9c25a2bd9a0cc4fcc97187a91e15

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx (hunk 5 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/AddEditMGLabel.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8516/commits/c285a00fc8cc9c25a2bd9a0cc4fcc97187a91e15

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/DeleteMGLabel.jsx (hunk 6 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/MicrogatewayLabels/DeleteMGLabel.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8516/commits/b0610c2fe8e3696091bd4f302f76ed2d54c616f4

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx (hunk 4 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8516/commits/b0610c2fe8e3696091bd4f302f76ed2d54c616f4

**Comment by chathuranga95** [comment]
### Please find the latest changes to the Add-Edit MG label component.

enhanced mg label popup

**Pull request merged**